send/ldap_it4i

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/perl
+use strict;
+use warnings;
+no if $] >= 5.017011, warnings => 'experimental::smartmatch';
+use Net::LDAPS;
+use Net::LDAP::Entry;
+use Net::LDAP::Message;
+use Net::LDAP::LDIF;
+use ScriptLock;
+use ADConnector;
+
+sub process_update;
+sub process_archive;
+
+# Constants and counters
+my $counter_add = 0;
+my $counter_update = 0;
+my $counter_archive = 0;
+my $counter_fail = 0;
+
+# Define service
+my $service_name = "ldap_it4i";
+
+# GEN folder location
+my $facility_name = $ARGV[0];
+chomp($facility_name);
+my $service_files_base_dir="../gen/spool";
+my $service_files_dir="$service_files_base_dir/$facility_name/$service_name";
+my $service_file = "$service_files_dir/$service_name.ldif";
+
+# BASE DN
+open my $file, '<', "$service_files_dir/baseDN";
+my $base_dn = <$file>;
+chomp($base_dn);
+close $file;
+
+# propagation destination
+my $namespace = $ARGV[1];
+chomp($namespace);
+
+# create service lock
+my $lock = ScriptLock->new($facility_name . "_" . $service_name . "_" . $namespace);
+($lock->lock() == 1) or die "Unable to get lock, service propagation was already running.";
+
+# init configuration
+my @conf = init_config($namespace);
+# we cannot use a function to resolve PDC from DNS since the LDAP instance is not an AD, therefore, we use the raw value
+my $ldap_location = $conf[0];
+my $ldap = ldap_connect($ldap_location);
+my $filter = '(objectClass=person)';
+
+# connect
+ldap_bind($ldap, $conf[1], $conf[2]);
+
+# load all data
+my @users = load_perun($service_file);
+my @previousUsers = load_ad($ldap, $base_dn, $filter, ['uid','givenName','sn','fullName','displayName','mail','userPassword','state','sshPublicKey']);
```

Comment:
Based on other comments, there should be `fullName` replaced with `cn` and `state` with `status`. Also we probably do not need to get/update `userPassword`, since users shouldn't have their login changed! It would be considered as different entry by the script anyway.

Reply:
Fixed.